File: algo/src/BOJ/Gold/BOJ1083.java
package BOJ.Gold;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class BOJ1083 {
    public static void main(String[] args) throws IOException {
        BufferedReader br =new BufferedReader(new InputStreamReader(System.in));
        var n = Integer.parseInt(br.readLine());
        var arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        var m = Integer.parseInt(br.readLine());

        for (int i = 0; i < arr.length; i++) {
            int max = arr[i];
            for (int j = 0; j < arr.length; j++) {

            }
        }
        Arrays.stream(arr).forEach(value -> System.out.print(value+" "));
    }
}

File: algo/src/BOJ/Gold/BOJ1083.kt
package BOJ.Gold

fun main() {
    val n = readln().toInt()
    val arr = readln().split(" ").map(String::toInt).toCollection(ArrayList())
    var s = readln().toInt()

    for (i in arr.indices) {
        if(s==0)break
        var flag = false
        var changeIndex =i
        for( j in i until arr.size){
            if(arr[i]<arr[j]){
                if(j-i<=s&&arr[changeIndex]<arr[j]){
                    flag =true
                   changeIndex =j
                }
            }
        }
        //스왑
        if (flag){
            val num = arr[changeIndex]
            arr.removeAt(changeIndex)
            arr.add(i,num)
            s-=changeIndex-i
        }
    }
    arr.forEach {
        print("$it ")
    }
}